core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -64,7 +69,7 @@ private ManifestWriter(PartitionSpec spec, OutputFile file, Long snapshotId) {
   protected abstract ManifestEntry<F> prepare(ManifestEntry<F> entry);
 
   protected abstract FileAppender<ManifestEntry<F>> newAppender(
-      PartitionSpec spec, OutputFile outputFile);
+      PartitionSpec spec, OutputFile outputFile, String compressionCodec, Integer compressionLevel);
```

Comment:
If I do
```
  protected FileAppender<ManifestEntry<F>> newAppender(PartitionSpec spec, OutputFile outputFile) {
    return newAppender(spec, outputFile, null, null);
  }

  protected abstract FileAppender<ManifestEntry<F>> newAppender(
      PartitionSpec spec, OutputFile outputFile, String compressionCodec, Integer compressionLevel);
```
I get
```
> There were Java public API/ABI breaks reported by revapi:
  
  java.method.abstractMethodAdded: Abstract method was added.
  
  old: <none>
  new: method org.apache.iceberg.io.FileAppender<org.apache.iceberg.ManifestEntry<F>> org.apache.iceberg.ManifestWriter<F extends org.apache.iceberg.ContentFile<F extends org.apache.iceberg.ContentFile<F>>>::newAppender(org.apache.iceberg.PartitionSpec, org.apache.iceberg.io.OutputFile, java.lang.String, java.lang.Integer)
  
```

Reply:
I believe it should be as I outlined in https://github.com/apache/iceberg/commit/1b237d570197d52fc0c7d959abc28ff14830d9aa. After one minor release we can then clean up the API.